torch/nn/modules/activation.py

```diff
@@ -1061,7 +1061,19 @@ def forward(self, query: Tensor, key: Tensor, value: Tensor, key_padding_mask: O
         .. note::
             `batch_first` argument is ignored for unbatched inputs.
         """
-        is_batched = query.dim() == 3
+        is_batched = True
+        qsize = query.size()
+        ksize = key.size()
+        vsize = value.size()
+        if self.batch_first:
+            query = query.view(-1, qsize[-2], qsize[-1])
+            key = key.view(-1, ksize[-2], ksize[-1])
+            value = value.view(-1, vsize[-2], vsize[-1])
+        else:
+            query = query.view(qsize[0], qsize[1], -1)
```

Comment:
this doesn't look right, for non-batch-first sizes are (S,N,E) or (S,E), so it should be `.view(size(0), -1, size(-1)`